src/key/key.js

```diff
@@ -498,6 +500,30 @@ Key.prototype.clearPrivateParams = function () {
   });
 };
 
+/**
+ * Remove private primary key material, converting the key to a dummy one
+ * The resulting primary key cannot be used for signing/decrypting but can still verify the subkeys
+ * @returns {<module:key.Key} dummy version of the key
+ */
+Key.prototype.toDummy = function () {
+  if (this.primaryKey.isDummy()) {
+    return;
+  }
+  if (!this.isPrivate()) {
+    throw new Error("Cannot convert a public key");
+  }
+  if (!this.primaryKey.isDecrypted()) {
+    // this is technically not needed, but makes the conversion simpler
+    throw new Error("Key is not decrypted");
+  }
+  this.primaryKey.clearPrivateParams();
+  this.primaryKey.isEncrypted = false;
+  this.primaryKey.s2k = new S2K();
+  this.primaryKey.s2k.algorithm = 0;
+  this.primaryKey.s2k.c = 0;
+  this.primaryKey.s2k.type = 'gnu-dummy';
+};
+
```

Comment:
I think this should be a method on `packet.SecretKey.prototype` (so that, in theory, it can also be used on subkeys), and be called `makeDummy`. From `Key.prototype.toDummy` I would expect to make a clone of the key, for consistency with `toPublic`, however we don't necessarily have to add that in this PR (also, it doesn't make clear that it only makes the primary key dummy, so maybe the name is still not ideal, cf. GNU `--export-secret-subkeys`).